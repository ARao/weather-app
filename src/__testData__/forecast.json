{
    "location": {},
    "current": {
        "condition": {}
    },
    "forecast": {
        "forecastday": [
            {
                "day": {
                    "condition": {}
                },
                "astro": {}
            },
            {
                "day": {
                    "condition": {}
                },
                "astro": {
                }
            }
        ]
    }
}