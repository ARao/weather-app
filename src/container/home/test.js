import React from 'react';
import { shallow } from 'enzyme';
import ConnectdHome from './index';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import toJson from 'enzyme-to-json'
import current from '../../__testData__/current.json'
import forecast from '../../__testData__/forecast.json'
// import { fetchHistories } from '../../actions/historyActions'



describe('Home connectd component', () => {
  const fetchCurrent = jest.fn();
  const fetchForecast = jest.fn();
  const middleware = [thunk]

  const initialState = {
    weather: {
      forecast,
      current,
    }
  }
  const mockStore = configureStore(middleware);
  let wrapper, store;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<ConnectdHome store={store} fetchCurrent={fetchCurrent} fetchForecast={fetchForecast} />)
  })

  it('++++ render the connected(SMART) component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('++++ store should match with value currentWeather', () => {
    expect(wrapper.props().currentWeather).toEqual(initialState.weather.current)
  });

  it('++++ store should match with value forecastWeather', () => {
    expect(wrapper.props().forecastWeather).toEqual(initialState.weather.forecast)
  });

  it('+++ snapshot match', () => {
    const snap = toJson(wrapper.dive())
    expect(snap).toMatchSnapshot();
  });
});
