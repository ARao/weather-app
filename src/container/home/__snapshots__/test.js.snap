// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home connectd component +++ snapshot match 1`] = `
<Fragment>
  <div
    className="row"
  >
    <div
      className="mr-auto"
    >
      <h1>
        Weather
      </h1>
    </div>
  </div>
  <BasicInfo
    weather={
      Object {
        "current": Object {
          "condition": Object {},
        },
        "location": Object {},
      }
    }
  />
  <div
    className="row"
  >
    <div
      className="ml-auto"
    >
      <TempInfo
        weather={
          Object {
            "condition": Object {},
          }
        }
      />
    </div>
  </div>
  <div
    className="row"
  >
    <DayCard
      day={
        Object {
          "astro": Object {},
          "day": Object {
            "condition": Object {},
          },
        }
      }
      index={0}
      key="Daycard--0"
    />
    <DayCard
      day={
        Object {
          "astro": Object {},
          "day": Object {
            "condition": Object {},
          },
        }
      }
      index={1}
      key="Daycard--1"
    />
  </div>
  <Link
    replace={false}
    to="history"
  >
    <button
      className="btn btn-success"
    >
       History 
    </button>
  </Link>
</Fragment>
`;
